File: go/key_test.go
// tests that correspond to the `src/key.rs` rust api
package main

import (
	"testing"

	"github.com/n0-computer/iroh-ffi/iroh"
	"github.com/stretchr/testify/assert"
)

// TestPublicKey tests all the constructors and methods of the PublicKey
func TestPublicKey(t *testing.T) {
	keyStr := "ki6htfv2252cj2lhq3hxu4qfcfjtpjnukzonevigudzjpmmruxva"
	fmtStr := "ki6htfv2252cj2lh"
	bytes := []byte("\x52\x3c\x79\x96\xba\xd7\x74\x24\xe9\x67\x86\xcf\x7a\x72\x05\x11\x53\x37\xa5\xb4\x56\x5c\xd2\x55\x06\xa0\xf2\x97\xb1\x91\xa5\xea")

	// create key from string
	key, err := iroh.PublicKeyFromString(keyStr)
	assert.Nil(t, err)

	// test methods are as expected
	assert.Equal(t, key.ToString(), keyStr)
	assert.Equal(t, key.ToBytes(), bytes)
	assert.Equal(t, key.FmtShort(), fmtStr)

	//create key from bytes
	key0, err := iroh.PublicKeyFromBytes(bytes)
	assert.Nil(t, err)

	// test methods are as expected
	assert.Equal(t, key0.ToString(), keyStr)
	assert.Equal(t, key0.ToBytes(), bytes)
	assert.Equal(t, key0.FmtShort(), fmtStr)

	// test eq method works
	assert.True(t, key.Equal(key0))
	assert.True(t, key0.Equal(key))

}

File: js/key.test.js
const iroh = require('../index');

test('public key', () => {
  const key_str = "ki6htfv2252cj2lhq3hxu4qfcfjtpjnukzonevigudzjpmmruxva";
  const fmt_str = "ki6htfv2252cj2lh";
  const bytes = [0x52, 0x3c, 0x79, 0x96, 0xba, 0xd7, 0x74, 0x24, 0xe9, 0x67, 0x86, 0xcf, 0x7a, 0x72, 0x05, 0x11, 0x53, 0x37, 0xa5, 0xb4, 0x56, 0x5c, 0xd2, 0x55, 0x06, 0xa0, 0xf2, 0x97, 0xb1, 0x91, 0xa5, 0xea];

  // create key from string
  const key = iroh.PublicKey.fromString(key_str);

  // test methods are as expected
  expect(key.toString()).toBe(key_str);
  expect(key.toBytes()).toEqual(bytes);
  expect(key.fmtShort()).toBe(fmt_str);

  // create key from bytes
  const key_0 = iroh.PublicKey.fromBytes(bytes);

  // test methods are as expected
  expect(key_0.toString()).toBe(key_str);
  expect(key_0.toBytes()).toEqual(bytes);
  expect(key_0.fmtShort()).toBe(fmt_str);

  // test that the equal function works
  expect(key.equal(key_0)).toBe(true);
  expect(key_0.equal(key)).toBe(true);
})
